### PR TITLE
Update server foundation api references

### PR DESCRIPTION
**Pull request by elgnay**
Update out-of-date api references owned by server foundation. See https://github.com/open-cluster-management/backlog/issues/6498 for more details.

### CODE REVIEW

**Review by cadawson — COMMENTED** [review]
This looks good to me. Thank you @elgnay . If @qiujian16 approves, I am fine with merging these.

**Comment by qiujian16** [comment]
@elgnay @cadawson I think we should drop some sentences here to indicate they are alpha apis.

**Comment by elgnay** [comment]
> @elgnay @cadawson I think we should drop some sentences here to indicate they are alpha apis.

I added the api version into the section title for clusterset and clustersetbinding.

File: apis/clusterset.json.adoc [hunk 1]
```diff
@@ -0,0 +1,254 @@
...
+= Clustersets API (v1alpha1)
+
+[[_rhacm-docs_apis_clusterset_jsonoverview]]
+== Overview
+This documentation is for the clusterset resource for {product-title}. Clusterset resource has four possible requests: create, query, delete and update.
+
+
+=== Version information
+[%hardbreaks]
+__Version__ : 2.0.0
```
Comment (dockerymick) [inline apis/clusterset.json.adoc]: Update the version to 2.1.0

File: apis/clustersetbinding.json.adoc [hunk 2]
```diff
@@ -0,0 +1,278 @@
...
+= Clustersetbindings API (v1alpha1)
+
+[[_rhacm-docs_apis_clustersetbinding_jsonoverview]]
+== Overview
+This documentation is for the clustersetbinding resource for {product-title}. Clustersetbinding resource has four possible requests: create, query, delete and update.
+
+
+=== Version information
+[%hardbreaks]
+__Version__ : 2.0.0
```
Comment (dockerymick) [inline apis/clustersetbinding.json.adoc]: Update the version here

File: apis/clusterset.json.adoc [hunk 3]
```diff
@@ -0,0 +1,254 @@
...
+
+
+[[_rhacm-docs_apis_clusterset_jsoncreateclusterset]]
+=== Create a clusterset
+....
+POST /cluster.open-cluster-management.io/v1alpha1/managedclustersets
+....
+
+
+==== Description
```
Comment (dockerymick) [inline apis/clusterset.json.adoc:99]: add punctuation here.

File: apis/clusterset.json.adoc [hunk 4]
```diff
@@ -0,0 +1,254 @@
...
+[[_rhacm-docs_apis_clusterset_jsondeleteclusterset]]
+=== Delete a clusterset
+....
+DELETE /cluster.open-cluster-management.io/v1alpha1/managedclustersets/{clusterset_name}
+....
+
+
+==== Description
+Delete a single clusterset
+
```
Comment (dockerymick) [inline apis/clusterset.json.adoc:205]: Add punctuation

File: apis/clustersetbinding.json.adoc [hunk 5]
```diff
@@ -0,0 +1,278 @@
...
+
+
+[[_rhacm-docs_apis_clustersetbinding_jsoncreateclustersetbinding]]
+=== Create a clustersetbinding
+....
+POST /cluster.open-cluster-management.io/v1alpha1/namespaces/{namespace}/managedclustersetbindings
+....
+
+
+==== Description
```
Comment (dockerymick) [inline apis/clustersetbinding.json.adoc:101]: Add punctuation

File: apis/clustersetbinding.json.adoc [hunk 6]
```diff
@@ -0,0 +1,278 @@
...
+
+[[_rhacm-docs_apis_clustersetbinding_jsondeleteclustersetbinding]]
+=== Delete a clustersetbinding
+....
+DELETE /cluster.open-cluster-management.io/v1alpha1/managedclustersetbindings/{clustersetbinding_name}
+....
+
+
+==== Description
+Delete a single clustersetbinding
```
Comment (dockerymick) [inline apis/clustersetbinding.json.adoc]: Add punctuation

**Review by dockerymick — COMMENTED** [review]
made a couple of comments about the descriptions and version numbers.

File: apis/clusterset.json.adoc (hunk 1 above)
Comment (elgnay) [reply apis/clusterset.json.adoc]: Updated.

File: apis/clustersetbinding.json.adoc (hunk 2 above)
Comment (elgnay) [reply apis/clustersetbinding.json.adoc]: Updated.

File: apis/clusterset.json.adoc (hunk 3 above)
Comment (elgnay) [reply apis/clusterset.json.adoc:99]: Done.

File: apis/clusterset.json.adoc (hunk 4 above)
Comment (elgnay) [reply apis/clusterset.json.adoc:205]: Done.

File: apis/clustersetbinding.json.adoc (hunk 5 above)
Comment (elgnay) [reply apis/clustersetbinding.json.adoc:101]: Done.

File: apis/clustersetbinding.json.adoc (hunk 6 above)
Comment (elgnay) [reply apis/clustersetbinding.json.adoc]: Done.

**Pull request merged**